File: Plugins/AM_DeconPeakDetector_PlugIn/clsAnalysisResourcesDeconPeakDetector.vb
'*********************************************************************************************************
' Written by Matthew Monroe for the US Department of Energy 
' Pacific Northwest National Laboratory, Richland, WA
' Created 05/23/2014
'
'*********************************************************************************************************

Option Strict On

Imports AnalysisManagerBase
Imports System.IO

Public Class clsAnalysisResourcesDeconPeakDetector
	Inherits clsAnalysisResources
	
	Public Overrides Function GetResources() As IJobParams.CloseOutType

		Dim strRawDataType As String = m_jobParams.GetJobParameter("RawDataType", "")
		
		' Retrieve the peak detector parameter file

		Dim peakDetectorParamFileName = m_jobParams.GetJobParameter("PeakDetectorParamFile", "")
		Dim paramFileStoragePath = m_jobParams.GetParam("ParmFileStoragePath")

		paramFileStoragePath = Path.Combine(paramFileStoragePath, "PeakDetection")

		If Not RetrieveFile(peakDetectorParamFileName, paramFileStoragePath) Then
			Return IJobParams.CloseOutType.CLOSEOUT_FAILED
		End If

		' Retrieve the instrument data file
		If Not RetrieveSpectra(strRawDataType) Then
			If String.IsNullOrEmpty(m_message) Then
				m_message = "Error retrieving instrument data file"
			End If

			clsLogTools.WriteLog(clsLogTools.LoggerTypes.LogFile, clsLogTools.LogLevels.DEBUG, "clsDtaGenResources.GetResources: " & m_message)
			Return IJobParams.CloseOutType.CLOSEOUT_FAILED
		End If

		If Not MyBase.ProcessMyEMSLDownloadQueue(m_WorkingDir, MyEMSLReader.Downloader.DownloadFolderLayout.FlatNoSubfolders) Then
			Return IJobParams.CloseOutType.CLOSEOUT_FAILED
		End If

		Return IJobParams.CloseOutType.CLOSEOUT_SUCCESS

	End Function

End Class

File: Plugins/AM_DeconPeakDetector_PlugIn/clsAnalysisResourcesDeconPeakDetector.cs
//*********************************************************************************************************
// Written by Matthew Monroe for the US Department of Energy
// Pacific Northwest National Laboratory, Richland, WA
// Created 05/23/2014
//
//*********************************************************************************************************

using AnalysisManagerBase;
using System.IO;

namespace AnalysisManagerDeconPeakDetectorPlugIn
{

    /// <summary>
    /// Retrieve resources for the Decon Peak Detector plugin
    /// </summary>
    public class clsAnalysisResourcesDeconPeakDetector : clsAnalysisResources
    {

        /// <summary>
        /// Retrieve required files
        /// </summary>
        /// <returns>Closeout code</returns>
        public override CloseOutType GetResources()
        {
            // Retrieve shared resources, including the JobParameters file from the previous job step
            var result = GetSharedResources();
            if (result != CloseOutType.CLOSEOUT_SUCCESS)
            {
                return result;
            }

            var strRawDataType = mJobParams.GetJobParameter("RawDataType", "");

            // Retrieve the peak detector parameter file

            var peakDetectorParamFileName = mJobParams.GetJobParameter("PeakDetectorParamFile", "");
            var paramFileStoragePath = mJobParams.GetParam("ParmFileStoragePath");

            paramFileStoragePath = Path.Combine(paramFileStoragePath, "PeakDetection");

            if (!FileSearch.RetrieveFile(peakDetectorParamFileName, paramFileStoragePath))
            {
                return CloseOutType.CLOSEOUT_FAILED;
            }

            // Retrieve the instrument data file
            if (!FileSearch.RetrieveSpectra(strRawDataType))
            {
                if (string.IsNullOrEmpty(mMessage))
                {
                    mMessage = "Error retrieving instrument data file";
                }

                LogDebug("clsDtaGenResources.GetResources: " + mMessage);
                return CloseOutType.CLOSEOUT_FAILED;
            }

            if (!base.ProcessMyEMSLDownloadQueue(mWorkDir, MyEMSLReader.Downloader.DownloadLayout.FlatNoSubdirectories))
            {
                return CloseOutType.CLOSEOUT_FAILED;
            }

            return CloseOutType.CLOSEOUT_SUCCESS;
        }
    }
}
